cli/lsp/tsc.rs

```diff
@@ -817,35 +925,60 @@ impl TsServer {
     snapshot: Arc<StateSnapshot>,
     specifier: ModuleSpecifier,
     position: u32,
-  ) -> Result<Option<Vec<ImplementationLocation>>, LspError> {
+  ) -> Result<Option<Vec<ImplementationLocation>>, AnyError> {
     let req = TscRequest::GetImplementationAtPosition((
       self.specifier_map.denormalize(&specifier),
       position,
     ));
-    self
-      .request::<Option<Vec<ImplementationLocation>>>(snapshot, req)
-      .await
-      .and_then(|mut locations| {
-        for location in locations.iter_mut().flatten() {
-          location.normalize(&self.specifier_map)?;
-        }
-        Ok(locations)
-      })
-      .map_err(|err| {
-        log::error!("Failed to request to tsserver {}", err);
-        LspError::invalid_request()
-      })
+    let mut results = FuturesOrdered::new();
+    for scope in snapshot
+      .config
+      .tree
+      .data_by_scope()
+      .keys()
+      .map(Some)
+      .chain(std::iter::once(None))
+    {
+      results.push_back(self.request::<Option<Vec<ImplementationLocation>>>(
+        snapshot.clone(),
+        req.clone(),
+        scope.cloned(),
+      ));
+    }
+    let mut all_locations = IndexSet::new();
+    while let Some(locations) = results.next().await {
+      let locations = locations
+        .inspect_err(|err| {
+          let err = err.to_string();
+          if !err.contains("Could not find source file") {
+            lsp_warn!("Unable to get implementations from TypeScript: {err}");
+          }
+        })
+        .unwrap_or_default();
+      let Some(mut locations) = locations else {
+        continue;
+      };
+      for location in &mut locations {
+        location.normalize(&self.specifier_map)?;
+      }
+      all_locations.extend(locations);
+    }
+    if all_locations.is_empty() {
+      return Ok(None);
```

Comment:
I wonder why this returns an option. Is something relying on this returning `None`? Maybe we can just return a vec now? If not, it looks like the behaviour might have changed?

Reply:
It's just preserving the TS API (https://github.com/denoland/deno/blob/v1.37.1/cli/tsc/dts/typescript.d.ts#L6228), which probably tries to match the LSP API. With all of our intermediate plumbing it's basically meaningless.